src/scripts/main.js

```diff
@@ -1,3 +1,42 @@
 'use strict';
 
-// write code here
+const grid = document.querySelector('table');
+
+grid.addEventListener('click', (e) => {
+  if (e.target.tagName !== 'TH') {
+    return;
+  }
+
+  const body = document.querySelector('tbody');
+  const rows = [...body.rows];
+  const collIndex = e.target.cellIndex;
+
+  const firstCell = rows[0].cells[collIndex].textContent.trim();
+  const type = !isNaN(firstCell.replace(/[,]/g, '')) ? 'number' : 'string';
```

Comment:
The check for numeric type using `isNaN(firstCell.replace(/[,]/g, ''))` might not handle all numeric formats correctly, especially if the numbers include currency symbols or other non-numeric characters. Consider using a more robust method to determine if the content is numeric.